src/js/messaging/containers/Main.jsx

```diff
@@ -35,13 +41,66 @@ class Main extends React.Component {
     this.props.setNewFolderName(field);
   }
 
-  handleSubmit() {}
+  handleCreateNewFolderModal() {
+    // Reset the form / state object when closed
+    const resetForm = { value: '', dirty: false };
+    this.props.setNewFolderName(resetForm);
+    this.props.toggleCreateFolderModal();
+  }
+
+  handleSubmit(domEvent) {
```

Comment:
This should probably be renamed to `handleSubmitCreateFolder` or something now to be more specific.